server/sonar-web/src/main/js/apps/sessions/components/LoginFormContainer.tsx

```diff
@@ -22,32 +22,41 @@ import * as PropTypes from 'prop-types';
 import { connect } from 'react-redux';
 import LoginForm from './LoginForm';
 import { doLogin } from '../../../store/rootActions';
-import { getAppState } from '../../../store/rootReducer';
+import { fetchAppState } from '../../../store/rootActions';
 import { IdentityProvider, getIdentityProviders } from '../../../api/users';
 import { getBaseUrl } from '../../../helpers/urls';
 
 interface Props {
   doLogin: (login: string, password: string) => Promise<void>;
+  fetchAppState: () => Promise<any>;
   location: { hash?: string; pathName: string; query: { return_to?: string } };
 }
 
 interface State {
   identityProviders?: IdentityProvider[];
+  onSonarCloud: boolean;
 }
 
 class LoginFormContainer extends React.PureComponent<Props, State> {
   mounted: boolean;
-  state: State = {};
+  state: State = { onSonarCloud: false };
 
   static contextTypes = {
     router: PropTypes.object
   };
 
   componentDidMount() {
     this.mounted = true;
-    getIdentityProviders().then(r => {
+    Promise.all([
+      getIdentityProviders(),
+      this.props.fetchAppState()
```

Comment:
When forced authentication is enabled, this WS fails with 401. In this case you need to avoid calling it, and assume that it's usual SonarQube on-premise.

Reply:
:thinking: I didn't think about that... I'm not sure how we could avoid calling it. Because unless it fails we don't know if there is forced authentication...
But I could still call it and prevent it from raising an error and consider that we are on-premise as usual.